transformer_engine/pytorch/module/grouped_linear.py

```diff
@@ -153,12 +170,14 @@ def forward(
 
             _ = fp8_grouped_gemm(
                 [w._data for w in weights_fp8],
-                fp8_meta["scaling_fwd"].scale_inv,
-                _GEMM_WEIGHT,
+                torch.cat(
+                    [w._scale_inv for w in weights_fp8]
+                ),  # avoiding torch.cat requires another interface,
+                0,  # weight offset is 0 for the newly created _scale_inv
```

Comment:
Hi @timmoon10, I want to confirm this is a bug, right? FP8 weights are casted in `get_fp8_workspace` in which a standalone `scale_inv` is set for each Float8Tensor instead of using `fp8_meta["scaling_fwd"].scale_inv`.

I think it's a bit confusing that in `get_fp8_workspace`:
- `with_transpose`: Only `Float8Tensor._scale_inv` is updated.
- not `with_transpose`: `fp8_meta.scale_inv[fp8_meta_index]` is updated and then copied to `Float8Tensor._scale_inv`.
https://github.com/NVIDIA/TransformerEngine/blob/47caafb28fd3f8c99ada92f9d7504715d268aa91/transformer_engine/pytorch/module/base.py#L997